101-aks/azuredeploy.json

```diff
@@ -0,0 +1,206 @@
+			}
+		},
+		"dnsPrefix": {
+			"type": "string",
+			"metadata": {
+				"description": "Optional DNS prefix to use with hosted Kubernetes API server FQDN."
+			}
+		},
+		"osDiskSizeGB": {
+			"type": "int",
+			"defaultValue": 0,
+			"metadata": {
+				"description": "Disk size (in GB) to provision for each of the agent pool nodes. This value ranges from 0 to 1023. Specifying 0 will apply the default disk size for that agentVMSize."
+			},
+			"minValue": 0,
+			"maxValue": 1023
+		},
+		"agentCount": {
+			"type": "int",
+			"defaultValue": 3,
+			"metadata": {
+				"description": "The number of nodes for the cluster."
+			},
+			"minValue": 1,
+			"maxValue": 50
+		},
+		"agentVMSize": {
+			"type": "string",
+			"defaultValue": "Standard_D2_v2",
+			"allowedValues": [
+				"Standard_A10",
+				"Standard_A11",
+				"Standard_A2",
+				"Standard_A2_v2",
+				"Standard_A2m_v2",
+				"Standard_A3",
+				"Standard_A4",
+				"Standard_A4_v2",
+				"Standard_A4m_v2",
+				"Standard_A5",
+				"Standard_A6",
+				"Standard_A7",
+				"Standard_A8",
+				"Standard_A8_v2",
+				"Standard_A8m_v2",
+				"Standard_A9",
+				"Standard_D11",
+				"Standard_D11_v2",
+				"Standard_D12",
+				"Standard_D12_v2",
+				"Standard_D13",
+				"Standard_D13_v2",
+				"Standard_D14",
+				"Standard_D14_v2",
+				"Standard_D15_v2",
+				"Standard_D2",
+				"Standard_D2_v2",
+				"Standard_D3",
+				"Standard_D3_v2",
+				"Standard_D4",
+				"Standard_D4_v2",
+				"Standard_D5_v2",
+				"Standard_DS11",
+				"Standard_DS11_v2",
+				"Standard_DS12",
+				"Standard_DS12_v2",
+				"Standard_DS13",
+				"Standard_DS13_v2",
+				"Standard_DS14",
+				"Standard_DS14_v2",
+				"Standard_DS15_v2",
+				"Standard_DS2",
+				"Standard_DS2_v2",
+				"Standard_DS3",
+				"Standard_DS3_v2",
+				"Standard_DS4",
+				"Standard_DS4_v2",
+				"Standard_DS5_v2",
+				"Standard_F16",
+				"Standard_F16s",
+				"Standard_F2",
+				"Standard_F2s",
+				"Standard_F4",
+				"Standard_F4s",
+				"Standard_F8",
+				"Standard_F8s",
+				"Standard_G1",
+				"Standard_G2"
+			],
+			"metadata": {
+				"description": "The size of the Virtual Machine."
+			}
+		},
+		"linuxAdminUsername": {
+			"type": "string",
+			"defaultValue": "azureuser",
+			"metadata": {
+				"description": "User name for the Linux Virtual Machines."
+			}
+		},
+		"sshRSAPublicKey": {
+			"type": "string",
+			"metadata": {
+				"description": "Configure all linux machines with the SSH RSA public key string. Your key should include three parts, for example 'ssh-rsa AAAAB...snip...UcyupgH azureuser@linuxvm'"
+			}
+		},
+		"servicePrincipalClientId": {
+			"metadata": {
+				"description": "Client ID (used by cloudprovider)"
+			},
+			"type": "securestring",
+			"defaultValue": "n/a"
+		},
+		"servicePrincipalClientSecret": {
+			"metadata": {
+				"description": "The Service Principal Client Secret."
+			},
+			"type": "securestring",
+			"defaultValue": "n/a"
+		},
+		"osType": {
```

Comment:
if there's only one allowed value then we can remove the param and use a literal or var

Reply:
Multiple OS types will be arriving soon.